library/core/src/ops/arith.rs

```diff
@@ -462,6 +462,10 @@ mul_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
 #[lang = "div"]
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_on_unimplemented(
+    on(
+        all(any(_Self = "f32", _Self = "f64",), Rhs = "{integer}",),
+        note = "consider using a floating-point literal"
```

Comment:
Thanks for the PR!  Cool that this can be done with `rustc_on_unimplemented`.

One tweak I'd request: can it also say some thing like "[...] by writing it with `.0`"?  Saying "floating-point literal" is good and accurate, giving them something to look up, but doesn't directly help someone who wouldn't already know how to fix this just from seeing `{integer}`.